README.md

```diff
@@ -286,52 +292,169 @@ type Action = {
+
+Verifications are stored in the Verifications Set, which is a modified [Last-Write-Wins-Element Set](<https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type#LWW-Element-Set_(Last-Write-Wins-Element-Set)>). Valid and unremoved `VerificationAdd` messages are stored in the `adds` set, indexed by their `claimHash`, a hash of the `VerificationClaim`. Any Farcaster account and external identity pair will have the same `claimHash`.
+
+Verifications can be removed by sending a valid `VerificationRemove` message to the network that references the `claimHash` to remove.
+
+```ts
+type VerificationRemove = {
+  message: {
+    body: {
+      claimHash: string;
+      schema: "farcaster.xyz/schemas/v1/verification-remove";
+    };
+    account: number;
+    timestamp: number;
+  };
+  envelope: {
+    hash: string;
+    hashType: "BLAKE2b";
+    signature: string;
+    signatureType: "ed25519" | "ecdsa-secp256k1";
+    signerPubKey: string;
+  };
+};
+```
+
+A valid `VerificationRemove` message is stored in the `removes` set, indexed by `claimHash`, and removes `claimHash` and the relevant `VerificationAdd` message from the `adds` set. A `claimHash` cannot be in both the `adds` set and the `removes` set, and conflicts are resolved using message timestamps.
+
+All `VerificationAdd` messages contain `externalAddressUri`, so a client can iterate through the `VerificationAdd` messages in the `adds` set to get all verified external identities. The `VerificationRemove` message schema only has `claimHash`, so the external identity is effectively hidden once the verification is removed.
+
+If a `VerificationAdd` or `VerificationRemove` message is received that matches a `claimHash` already in the `adds` or `removes` set, the message with a later timestamp wins. For example, if a `VerificationAdd` message is received whose `claimHash` matches one already in the `removes` set, and the timestamp of the `VerificationAdd` message is more recent than the `VerificationRemove` message in the `removes` set, then the `claimHash` is moved from the `removes` set to the `adds` set. Therefore, a verification between a Farcaster account and a particular external identity can be re-added if it was previously removed via a new `VerificationAdd` message.
+
+#### Message validation
+
+The `envelope` of `VerificationAdd` and `VerificationRemove` messages is validated the same as other message types. Here are unique rules for `VerificationAdd` messages:
+
+1. `schema` must be known
+2. `claimHash` must be present and match `hashFn(claim)` where `claim` is a `VerificationClaim` object made up of `externalAddressUri` and `account`
+3. `externalSignatureType` must be known
+4. `externalSignature` must be a valid [EIP 191 version 0x45](https://eips.ethereum.org/EIPS/eip-191) signature of the `claimHash` by `externalAddressUri`
+
+Here are rules for `VerificationRemove` messages:
+
+1. `schema` must be known
+2. `claimHash` must be present
+
+#### Set construction and pseudocode
```

Comment:
The `Set construction and pseudocode` section has a different level of detail than the `Set construction` section in other areas like `Actions`, for instance. The main difference is that this section goes into the data structures and the exact operations to update those structure, while the other one uses higher level descriptions. 

For instance, `If removes[claimHash].timestamp is more recent than add(claimHash).timestamp, reject the incoming message` vs `If x > y, discard y` 


The first point I'd make is that we should keep the language consistent so that people can read different sections with the same lens and have the same takeaways. 

The second point is that this section should simply express the logic as correctly as possible, while avoiding implementation details. This makes it easier for people to read, and they can always go to the code for a specific implementation.

Reply:
Good feedback, thank you. Updated the section to have a similar level of abstraction as others and made the pseudocode more readable.